src/core/aws-net-websocket-registry.adb

```diff
@@ -568,6 +572,24 @@ package body AWS.Net.WebSocket.Registry is
          Registered.Clear;
       end Finalize;
 
+      ----------
+      -- Free --
+      ----------
+
+      procedure Free (WebSocket : in out Object_Class) is
+      begin
+         --  If WebSocket is in Sending it means that it has been
+         --  taken by the Get_Socket call. We cannot free it now, we
+         --  record this socket to be freed as soon as it is released
+         --  (Release_Socket) call.
+
+         if Sending.Contains (WebSocket.Id) then
+            WebSocket.To_Free := True;
+         else
+            Unchecked_Free (WebSocket);
+         end if;
+      end Free;
+
       ----------------
       -- Get_Socket --
       ----------------
```

Comment:
Shouldn't Get_Socket also avoid sockets that are marked with To_Free=True ?

Reply:
No because we do return only socket not in Sending container and in this case if it was to be free it would have been already done in Release_Socket.